packages/strapi-connector-mongoose/lib/mount-models.js

```diff
@@ -189,6 +189,22 @@ module.exports = async ({ models, target }, ctx) => {
       };
     };
 
+    const parseComponentRef = el => {
+      if (el.ref instanceof mongoose.Types.ObjectId) {
+        return el.ref.toString();
+      } else {
+        return el.ref;
+      }
+    };
+
+    const parseDynamicZoneRef = el => {
+      if (el.ref instanceof mongoose.Types.ObjectId) {
+        return { id: el.ref.toString() };
```

Comment:
Why do we need to put the id within an object here and not for the parseComponentRef ?

Reply:
the dz needs to return an object to give the typename info
 ```js
{
  compo: 'id', // we now the type statically
  dz: { 
    __component: 'compo-type-uid',
    id: 'compo-id'
  }
}
 ```